src/test/java/org/kiwiproject/registry/eureka/util/EurekaTestDataHelper.java

```diff
@@ -0,0 +1,178 @@
+package org.kiwiproject.registry.eureka.util;
+
+import static java.util.concurrent.TimeUnit.MINUTES;
+import static javax.ws.rs.client.Entity.json;
+import static javax.ws.rs.core.HttpHeaders.CONTENT_TYPE;
+import static javax.ws.rs.core.MediaType.APPLICATION_JSON;
+import static javax.ws.rs.core.MediaType.APPLICATION_JSON_TYPE;
+import static org.awaitility.Awaitility.await;
+import static org.kiwiproject.base.KiwiStrings.f;
+import static org.kiwiproject.jaxrs.KiwiResponses.successful;
+import static org.kiwiproject.registry.eureka.config.EurekaRegistrationConfig.DEFAULT_LEASE_EXPIRATION_DURATION_SECONDS;
+import static org.kiwiproject.registry.eureka.config.EurekaRegistrationConfig.DEFAULT_LEASE_RENEWAL_INTERVAL_SECONDS;
+import static org.kiwiproject.registry.eureka.server.EurekaRegistryService.DEFAULT_DATA_CENTER_INFO_CLASS;
+import static org.kiwiproject.registry.eureka.server.EurekaRegistryService.DEFAULT_DATA_CENTER_NAME;
+import static org.kiwiproject.registry.eureka.server.EurekaRegistryService.LEASE_DURATION_IN_SECONDS;
+import static org.kiwiproject.registry.eureka.server.EurekaRegistryService.LEASE_RENEWAL_INTERVAL_IN_SECONDS;
+import static org.kiwiproject.test.jaxrs.JaxrsTestHelper.assertNoContentResponse;
+import static org.kiwiproject.test.jaxrs.JaxrsTestHelper.assertNotFoundResponse;
+import static org.kiwiproject.test.jaxrs.JaxrsTestHelper.assertOkResponse;
+
+import lombok.experimental.UtilityClass;
+import lombok.extern.slf4j.Slf4j;
+import org.glassfish.jersey.message.GZipEncoder;
+import org.kiwiproject.jaxrs.KiwiGenericTypes;
+import org.kiwiproject.registry.eureka.common.EurekaInstance;
+import org.kiwiproject.registry.eureka.common.EurekaResponseParser;
+import org.kiwiproject.registry.model.ServiceInstance;
+import org.kiwiproject.registry.util.ServiceInfoHelper;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.images.builder.ImageFromDockerfile;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import javax.ws.rs.client.ClientBuilder;
+import javax.ws.rs.core.Response;
+
+@UtilityClass
+@Slf4j
+public class EurekaTestDataHelper {
+
+    public static ImageFromDockerfile eurekaImage() {
+        return new ImageFromDockerfile()
+                .withFileFromClasspath("Dockerfile", "eureka-server/Dockerfile")
+                .withFileFromClasspath("eureka-server-1.10.17.war", "eureka-server/eureka-server-1.10.17.war")
```

Comment:
I really, _really_, _**really**_ don't want us to put a 22.1 MB WAR file in our repository. I don't much care if it takes longer to have to download something. We should also see about caching the downloads the way that the embedded Postgres and Consul test extensions do. That way you will incur the hit (at least when developing locally) one time and it can remain cached locally. And maybe we can use the GitHub cache action to cache it for CI builds as well.

Also see: https://docs.github.com/en/repositories/working-with-files/managing-large-files/about-large-files-on-github

See the "Distributing large binaries" section, though using the Releases for the testing WAR also seems pretty icky.

Reply:
Changed the docker file to download it and removed the war.  If we decide to keep Eureka long term then we can talk about creating the released image to use.